### DISCUSSION TITLE
Possible GAT implementation

### DISCUSSION THREAD

**Discussion by gudeh** [aimat-lab/gcnn_keras#29] [Q&A]
Hi there, I would like to congratulate the authors of this project, seems you are doing a great work!
I am considering to go for the implementation of a GAT network to reproduce a paper. I would like to ask you: 
1) Why shouldn't I use pytorch, since they seem to have a [primitive GAT available in pytorch](https://towardsdatascience.com/hands-on-graph-neural-networks-with-pytorch-pytorch-geometric-359487e221a8) itself.
2) Do you think it is possible for me to attempt an implementation using your GCNN in keras, attempting to reproduce [this work.](https://ieeexplore.ieee.org/document/8920342)

**Comment by PatReis** [comment]
Hallo, thanks.
To answer your questions: We know that pytorch geometric offers probably the largest collection of graph networks as far as I know and also a great community. Most STOA graph networks use or publish within pytorch geometric. So if you want to have a long-lifetime code basis, I can safely recommend using pytorch. 
This package is implemented in tensorflow keras, which means, if you are familiar with keras, you probably find the usage intuitive. 
Our goal was to use graph networks as keras-like as possible. That includes the functional API and ragged tensors. With tensorflow there is still interest in industry as tensorflow makes model deployment easier. 
The main focus in kgcnn are molecules and materials (for having batched graphs). We have however some examples for citation networks, but that is still somewhat cumbersome with keras.

Let me check your reference concerning 2. and also GAT and GATv2 models again.
